### PR TITLE
Use dev version of core-lightning for faster polling

### PR DESCRIPTION
**Pull request by elsirion**
Core-lightning is doing some slow polling in production builds (e.g. channel openings, chain tip, …). Dev builds are said to speed things up or at least allow us to set `--dev-*` options to reduce polling latency. This should speed up integration tests.

### CODE REVIEW

**Comment by elsirion** [comment]
Apparently I need an ACK from *somone else* with merge rights after protecting master after accidentally pushing to master again recently. @justinmoon 🥺

**Review by justinmoon — APPROVED** [review]
LGTM. Thanks for not pushing directly to master. I know you would never do such a thing!

**Pull request merged**
